src/main/java/io/pivotal/cfapp/service/LegacyWorkloadsService.java

```diff
@@ -30,9 +32,12 @@ public LegacyWorkloadsService(
         this.settings = settings;
     }
 
-    public Mono<List<AppDetail>> getLegacyApplications(String list) {
-        Set<String> stacks = Set.copyOf(Arrays.asList(list.split("\\s*,\\s*")));
-        return getLegacyApplications(LegacyPolicy.builder().stacks(stacks).build());
+    public Mono<List<AppDetail>> getLegacyApplications(WorkloadsFilter workloadsFilter) {
+        return getLegacyApplications(LegacyPolicy.builder().stacks(workloadsFilter.getStacks()).build());
+    }
+
+    public Mono<List<AppRelationship>> getLegacyApplicationRelationships(WorkloadsFilter workloadsFilter) {
```

Comment:
I think this method-naming makes sense.  However, the thing I think we are lacking is the ability to identify orphaned service instances that are also legacy service offerings.

Reply:
Yes, agree, we lack the ability to identify orphaned service instances. Maybe once all the apps are moved to a new service instance, we can list all service instances using /snapshot/detail/si and filter out based on service to see legacy service instances(a manual excel filter process/cf-hoover UI).

Reply:
Might come back to this in s subsequent effort. The logic already exists to identify and report on orphaned service instances; it's just a matter of applying a secondary filter.  Thanks!